hoot-core/src/main/cpp/hoot/core/OsmMap.cpp

```diff
@@ -933,6 +883,13 @@ bool OsmMap::validate(bool strict) const
 
 void OsmMap::visitRo(ElementVisitor& visitor) const
 {
+  visitNodesRo(visitor);
+  visitWaysRo(visitor);
+  visitRelationsRo(visitor);
+}
+
+void OsmMap::visitNodesRo(ElementVisitor& visitor) const
```

Comment:
Yeah, I've had the same conclusion with a lot of element related changes on this project...only three types and the number is bounded.  Sometimes in I've gone ahead and collapsed things down to handle types agnostically anyway for just less code to look at.  Great if you can do it...otherwise not worth spending a ton of time on.  thanks.